browser/brave_news/brave_news_tab_helper.cc

```diff
@@ -89,48 +102,89 @@ bool BraveNewsTabHelper::IsSubscribed(const FeedDetails& feed_details) {
 }
 
 bool BraveNewsTabHelper::IsSubscribed() {
-  for (const auto& feed : GetAvailableFeeds()) {
-    if (IsSubscribed(feed))
+  for (const auto& feed_url : GetAvailableFeedUrls()) {
+    if (IsSubscribed(feed_url)) {
       return true;
+    }
   }
   return false;
 }
 
-void BraveNewsTabHelper::ToggleSubscription(const FeedDetails& feed_details) {
-  bool subscribed = IsSubscribed(feed_details);
-  auto* publisher = controller_->publisher_controller()->GetPublisherForFeed(
-      feed_details.feed_url);
+std::string BraveNewsTabHelper::GetTitleForFeedUrl(const GURL& feed_url) {
+  auto* default_publisher =
+      controller_->publisher_controller()->GetPublisherForFeed(feed_url);
+  if (default_publisher) {
+    return default_publisher->publisher_name;
+  }
+
+  auto it = base::ranges::find_if(
```

Comment:
It would make it a little easier to understand this function if we comment the intention of each chunk (e.g. "Get title from possible matching known Publisher" / "Get title we've fetched already" / "Fetch title from feed url")